run/rails-cat_album/app/views/photos/index.html.erb

```diff
@@ -5,7 +5,7 @@
     <% @photos.each do |photo| %>
       <div class="card" style="width: 18rem;">
         <div class="card-body">
-          <img class="card-img-top" src="<%= url_for(photo.image) %>" alt="Card image cap">
+          <img class="card-img-top" src="<%= image_url photo %>" alt="Card image cap">
```

Comment:
Have you tried using [image_tag](https://api.rubyonrails.org/classes/ActionView/Helpers/AssetTagHelper.html#method-i-image_tag) here instead of the hardcoded `img` HTML tag to see if it will use the correct ActiveStorage `service` automatically? (If this works you should be able to remove `image_url`.)

Reply:
@quartzmo I've tried using image tag in this capacity `<img class="card-img-top" src="<%= image_url(url_for(photo.image)) %>"alt="Card image cap">`, I'm testing out image_tag with `<%= image_tag photo.image, size: "200x200" %>` and it seems to work in local development. I'm going to start the build and deploy to see if it works in production!

Reply:
> In the production website, when I tried to click on the photo URL, it was broken because it pointed to the active storage blob instead of the cloud storage URL.

If you want to download the image by clicking on it, I think maybe you want to add a Rails `link_to` helper around the Rails `image_tag` helper? As discussed here: https://stackoverflow.com/questions/43280001/rails-image-as-a-link